node/silkworm/db/util.cpp

```diff
@@ -103,15 +110,29 @@ std::optional<ByteView> find_value_suffix(mdbx::cursor& table, ByteView key, Byt
     return res;
 }
 
-void upsert_storage_value(mdbx::cursor& state_cursor, ByteView storage_prefix, ByteView location, ByteView value) {
-    if (find_value_suffix(state_cursor, storage_prefix, location)) {
-        state_cursor.erase();
-    }
-    value = zeroless_view(value);
-    if (!value.empty()) {
-        Bytes data{location};
-        data.append(value);
-        state_cursor.upsert(to_slice(storage_prefix), to_slice(data));
+void upsert_storage_value(mdbx::cursor& state_cursor, ByteView storage_prefix, ByteView location, ByteView new_value) {
+    static const auto build_db_value = [](const ByteView& loc, const ByteView& val) -> Bytes {
```

Comment:
The old code was simpler.

Reply:
I know **but was also less efficient** cause it forced an erase in any case (which causes MDBX to free a page and, eventually on update, reallocate a new one).
Here instead if old value matches new value no operation is performed and if not there is no additional seek of key for upsertion as it is performed on current record.